httpcore/_async/connection_pool.py

```diff
@@ -230,7 +235,16 @@ async def handle_async_request(self, request: Request) -> Response:
             timeouts = request.extensions.get("timeout", {})
             timeout = timeouts.get("pool", None)
             try:
-                connection = await status.wait_for_connection(timeout=timeout)
+                kwargs = {
+                    "timeout": timeout,
+                    "requests_awaiting_connection": self._requests_awaiting_connection,
+                    "pool_size": len(self._pool),
+                }
+
+                async with Trace("wait_for_connection", logger, None, kwargs):
+                    self._requests_awaiting_connection += 1
+                    connection = await status.wait_for_connection(timeout=timeout)
+                    self._requests_awaiting_connection -= 1
```

Comment:
Should `self._requests_awaiting_connection` be decremented if an exception (for instance, in case of Timeout) is raised in `await status.wait_for_connection(timeout=timeout)`?

PS The same is in the sync part.

Reply:
I have missed that, thank you